cypress/e2e/directory.spec.ts

```diff
@@ -0,0 +1,70 @@
+/// <reference types="cypress" />
+import '../support/commands';
+
+describe('directory infinite scroll', () => {
+  const PAGE_LIMIT = 20;
+  let totalPages = 0;
+
+  before(() => {
+    // clear Db
+    cy.truncateDatabase();
+
+    // fill DB with 100 users
+    cy.createUsers(100);
+
+    // get total pages for tests
+    cy.request('http://localhost:3000/api/users').then(response => {
+      totalPages = response.body.totalPages;
+    });
+  });
+
+  beforeEach(() => {
+    cy.visit('http://localhost:3000/directory');
+  });
+
+  function checkInfiniteScroll(pages: number, totalPages: number) {
+    for (let i = 1; i <= pages; i++) {
+      cy.get('[data-cy="user-container"]').children().as('ulChildren');
+      i < totalPages
+        ? cy.get('@ulChildren').should('have.length', PAGE_LIMIT * i)
+        : // last page not guaranteed to be full
+        cy.get('@ulChildren').should('have.length.within', PAGE_LIMIT * (i - 1), PAGE_LIMIT * i);
+
+      cy.intercept('GET', '/api/users?page=*', req => {
+        req.on('response', res => res.setDelay(500));
+      }).as('getUsers');
+      cy.scrollTo('bottom');
+
+      // last page won't have a spinner
+      if (i === totalPages) break;
+      // check spinner happened & removed
+      cy.get('[data-cy="loading"]'); //check if exists
+      cy.get('[data-cy="loading"]').should('not.exist'); // removed after data loaded
```

Comment:
nit: formatting, empty line after breaks/continues & same line comments should typically be avoided

```suggestion
      // last page won't have a spinner
      if (i === totalPages) break;

      // spinner should exist & be removed after request completes
      cy.get('[data-cy="loading"]');
      cy.get('[data-cy="loading"]').should('not.exist');
```  
Also, is this missing a `cy.wait` or something? How does it know when the request is complete?

Reply:
@devonzara Cypress will keep retrying their commands for 4 secs each before it declares it failed. That's why a `cy.wait` isn't required